streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -53,7 +54,7 @@
 import static org.apache.kafka.common.utils.Utils.getPort;
 import static org.apache.kafka.streams.processor.internals.InternalTopicManager.WINDOW_CHANGE_LOG_ADDITIONAL_RETENTION_DEFAULT;
 
-public class StreamPartitionAssignor implements PartitionAssignor, Configurable {
+public class StreamPartitionAssignor implements PartitionAssignor, Configurable, TaskIdToPartitionsProvider {
```

Comment:
@guozhangwang i'm not really sure what you mean in point 2 above? Are you suggesting that `TaskManager` pushes `cachedTasks` and `prevActiveTasks` on to `StreamPartitionAssignor`? So the interface between `TaskManager` and `StreamsPartitionAssignor` would also need to have `setCachedTasks(..)` and `setPreviousActivetTasks(...)`